src/test/java/org/tikv/common/apiversion/RequestKeyCodecTest.java

```diff
@@ -177,5 +176,48 @@ void testV2Codec(RequestKeyV2Codec v2) {
     decoded = v2.decodeRegion(region);
     assertEquals(start, decoded.getStartKey());
     assertEquals(ByteString.EMPTY, decoded.getEndKey());
+
+    // test region out of keyspace
```

Comment:
Suggest to add following test cases:
1. "" < start key < "r000". As the logics for empty and key < "r000" (e.g. "m") are not the same.
2. Ditto for end key.
3. Handle region with `start_key` < "r000" and `end_key` > "r001"

Rest LGTM~